design/how-ironic-works.md

```diff
@@ -120,22 +104,30 @@ will deploy successfully without issues.
 
 ## What connectivity is required?
 
-To boot the discovery and deployment image on the baremetal hardware:
+Access to the BMC can be via a routed IP network, however this may be
+less desirable than having it on the same L2 network as Ironic from a
+security standpoint.
 
-* DHCP
+When virtual media is used, the BMC needs to be on a network that
+allows it to reach the host serving the virtual media image.
+
+To boot the discovery and deployment image on the node, it will need
+access to the ironic host using:
+
+* DHCP (for IP assignment and PXE instructions)
 * TFTP (if iPXE is not natively supported by the network interfaces.)
 * HTTP in order to download kernel/ramdisk images via HTTP over a TCP
   connection.
 
-The discovery and deployment ramdisk image needs to be able to:
+The discovery and deployment ramdisk image needs to:
 
-* DHCP
-* Resolve DNS
+* DHCP (via the ironic host, for IP assignment and PXE instructions)
+* Resolve DNS (FIXME - also via the ironic host?)
```

Comment:
It doesn't need to hand out any DNS address, it can point to another DNS server. Dnsmasq is not exactly a fan of doing this, but it can be done.

Reply:
If the network isn't routed, the only DNS server we'll see is the one on the ironic host, right?

Reply:
I don't think DNS is strictly required for the IPA ramdisk to work?  We just need to be able to tell it what the ironic endpoint is?

We do need some DNS records to exist for the OCP cluster to come up, but that's not really related to ironic?

Not blocking but we might want to drop the DNS references from this doc in a future commit.

Reply:
IPA needs to be able to resolve the name in the URL for the image being deployed, right?

Reply:
Only if you use a name in the URL, currently we just use an IP in dev-scripts and it works just fine

Reply:
https://github.com/metalkube/dev-scripts/blob/master/07_deploy_masters.sh#L32 for example

Reply:
OK, I guess if we already have the RHCOS image downloaded locally we can use the IP.

Reply:
> OK, I guess if we already have the RHCOS image downloaded locally we can use the IP.

Yeah, my assumption is we'll want to either download or package the image so it's cached on the provisioning host/appliance, so any name resolution would only apply to the initial configuration of that host.